file: allow to create CephFS dataPools with predefined names (#9296)
rook/rook: Documentation/ceph-filesystem-crd.md

    replicated:
      size: 3
  dataPools:

>     - name: replicated-pool

Reviewer: pool in the name seems redundant, how about removing the pool suffix from the examples?
```suggestion
    - name: replicated
```

Author: but since it is for example, i just wanted to mention *pool* suffix for storage class. Otherwise, in storage class we will have pool name just as myfs-replicated. Having myfs-replicated-pool (with pool name replicated-pool) a bit more user friendly, haven't it?

Reviewer: In the storage class, the setting is: `pool: myfs-replicated-pool`, right? If the setting name is `pool`, isn't it clear it's a pool?

Author: Okay, you win :)

The pools allow all of the settings defined in the Pool CRD spec. For more details, see the [Pool CRD](ceph-pool-crd.md) settings. In the example above, there must be at least three hosts (size 3) and at least eight devices (6 data + 2 coding chunks) in the cluster.

* `metadataPool`: The settings used to create the filesystem metadata pool. Must use replication.

> * `dataPools`: The settings to create the filesystem data pools. Optionally, pool name can be specified with `name` field. If multiple pools are specified, Rook will add the pools to the filesystem. Assigning users or files to a pool is left as an exercise for the reader with the [CephFS documentation](http://docs.ceph.com/docs/master/cephfs/file-layouts/). The data pools can use replication or erasure coding. If erasure coding pools are specified, the cluster must be running with bluestore enabled on the OSDs.

Reviewer: ```suggestion
* `dataPools`: The settings to create the filesystem data pools. Optionally, a pool name can be specified with the `name` field to override the generated name. If multiple pools are specified, Rook will add the pools to the filesystem. Assigning users or files to a pool is left as an exercise for the reader with the [CephFS documentation](http://docs.ceph.com/docs/master/cephfs/file-layouts/). The data pools can use replication or erasure coding. If erasure coding pools are specified, the cluster must be running with bluestore enabled on the OSDs.
```

Author: > Optionally, a pool name can be specified with the `name` field to override the generated name.

How about:

Optionally, a pool name can be specified with the `name` field to override the default generated name.

> * `dataPools`: The settings to create the filesystem data pools. Optionally, a pool name can be specified with the `name` field to override the default generated name. If multiple pools are specified, Rook will add the pools to the filesystem. Assigning users or files to a pool is left as an exercise for the reader with the [CephFS documentation](http://docs.ceph.com/docs/master/cephfs/file-layouts/). The data pools can use replication or erasure coding. If erasure coding pools are specified, the cluster must be running with bluestore enabled on the OSDs.

Reviewer 2: I would like to be clearer here that while the `name` is optional and that we highly recommend setting the name to prevent issues that can arise from modifying the spec in a way that causes Rook to lose the original pool ordering. I would also suggest adding `name` as a field under `dataPools` here so that we can also document that the actual pool name stored in Ceph for each pool will be `<metadata.name>-<name>`. It will be easier to read the section just on `name` if it is separated. We can still mention here in the `dataPools` section that we recommend setting the `name` parameter on on pools that is documented below.

> * `dataPools`: The settings to create the filesystem data pools. Optionally, a pool name can be specified with the `name` field to override the default generated name, then final pool name will consist of filesystem name and pool name, eg. `<fsName>-<poolName>`. We are highly recommend to specify `name` to prevent issues that can arise from modifying the spec in a way that causes Rook to lose the original pool ordering. If multiple pools are specified, Rook will add the pools to the filesystem. Assigning users or files to a pool is left as an exercise for the reader with the [CephFS documentation](http://docs.ceph.com/docs/master/cephfs/file-layouts/). The data pools can use replication or erasure coding. If erasure coding pools are specified, the cluster must be running with bluestore enabled on the OSDs.

Reviewer 2: I think users will be able to read the docs more easily if the `name` field has its own section. This way, we can be clear about other features of `dataPools` without creating a paragraph that is cumbersome to understand.

```suggestion
* `dataPools`: The settings to create the filesystem data pools. Optionally (and we highly recommend), a pool name can be specified with the `name` field to override the default generated name; see more below.If multiple pools are specified, Rook will add the pools to the filesystem. Assigning users or files to a pool is left as an exercise for the reader with the [CephFS documentation](http://docs.ceph.com/docs/master/cephfs/file-layouts/). The data pools can use replication or erasure coding. If erasure coding pools are specified, the cluster must be running with bluestore enabled on the OSDs.
  * `name`: (optional, and highly recommended) Override the default generated name of the pool. The final pool name will consist of the filesystem name and pool name, e.g., `<fsName>-<poolName>`. We highly recommend to specify `name` to prevent issues that can arise from modifying the spec in a way that causes Rook to lose the original pool ordering. 
```

Author: Done
